dag.go

```diff
@@ -418,29 +386,73 @@ func setContextVariablesFromList(c *Config, r types.Resource, values []string, c
 	// all linked values should now have been processed as the graph
 	// will have handled them first
 	for _, value := range values {
+		// get the value from the linked resource
+		l, err := c.FindRelativeResource(value, r.Metadata().Module)
+		if err != nil {
+			return createParserError(
+				r,
+				fmt.Sprintf("unable to find dependent resource '%s': %s", value, err))
+		}
+
+		// can we set the context variables here?
+		var ctyRes cty.Value
+
+		// once we have found a resource convert it to a cty type and then
+		// set it on the context
+		switch l.Metadata().Type {
+		case resources.TypeLocal:
+			loc := l.(*resources.Local)
+			ctyRes = loc.CtyValue
+		case resources.TypeOutput:
+			out := l.(*resources.Output)
+			ctyRes = out.CtyValue
+		default:
+			ctyRes, err = convert.GoToCtyValue(l)
+		}
+
+		if err != nil {
+			return createParserError(
+				r,
+				fmt.Sprintf(`unable to convert reference %s to context variable: %s`, value, err))
+		}
+
+		// remove the attributes and to get a pure resource ref
+		// validate the name of the resource
 		fqrn, err := resources.ParseFQRN(value)
 		if err != nil {
-			pe := &errors.ParserError{}
-			pe.Filename = r.Metadata().File
-			pe.Line = r.Metadata().Line
-			pe.Column = r.Metadata().Column
-			pe.Message = fmt.Sprintf("error parsing resource link %s", err)
-			pe.Level = errors.ParserErrorLevelError
+			return createParserError(r, fmt.Sprintf("error parsing resource link %s", err))
+		}
+
+		fqrn.Attribute = ""
+
+		err = setContextVariableFromPath(ctx, fqrn.String(), ctyRes)
+		if err != nil {
+			return createParserError(r, fmt.Sprintf(`unable to set context variable: %s`, err))
+		}
+	}
+
+	return nil
+}
 
-			return pe
+// validateLinkedResources validates the linked resources in a resource
+// linked resources are extracted from the interpolated values in the resource
+// and are expected to be in the format of "module.module1.module2.resource.container.mine.id"
+
```

Comment:
do we need to add an additional // here so godoc connects that upper comment to the func too?

Reply:
This is not exported so really just for our info